src/thread_manager.ts

```diff
@@ -117,6 +118,19 @@ export class ThreadManager<SCG extends ExtendableGenerics = DefaultGenerics> {
     return () => unsubscribeFunctions.forEach((unsubscribe) => unsubscribe());
   };
 
+  private subscribeChannelDeleted = () => {
+    const unsubscribeFunctions = ['notification.channel_deleted', 'channel.deleted'].map((e) => {
```

Comment:
```suggestion
    const unsubscribeFunctions = ['notification.channel_deleted'].map((e) => {
```
One should suffice, notification-type events are always delivered (with less information) but in this case it's fine. With two in the array we're going to run the callback twice.

Reply:
Hey @arnautov-anton, notification events are not delivered for watched channels AFAIK. So we need to keep both.

Reply:
Just tested it just to make sure and I receive both notification and normal event even when the channel is watched though funny enough, I receive two notification events - I guess just to make 100% sure we handle it. :)

![image](https://github.com/user-attachments/assets/b112cfad-e3c5-4844-8324-c31b64a06309)